cachedcomplete/main_script.py

```diff
@@ -37,4 +41,14 @@ def _expand(filename):
 
 
 def get_files_to_hash():
+    '''
+    :return: an iterable of all the files and directories that should be hashed.
+    '''
     return chain([MAIN_FILE_PATH, os.path.dirname(__file__)], (_expand(filename) for filename in _get_info_list(FILES_TO_HASH_INFO)))
+
+
+def exists():
+    '''
+    :return: ``True`` if the main script exists, otherwise ``False`` (as when ran with `python -c`).
+    '''
+    return MAIN_FILE_PATH is not None and os.path.exists(MAIN_FILE_PATH)
```

Comment:
Instead of 
```python
MAIN_FILE_PATH is not None
``` 
you could simply
```python
MAIN_FILE_PATH
```

Reply:
I would argue `MAIN_FILE_PATH is not None` is more readable